app/src/main/java/org/mozilla/focus/widget/DownloadListAdapter.java

```diff
@@ -85,6 +85,11 @@ private void delete(int position){
         remove(position);
     }
 
+    private void cancel(int position){
+        DownloadManager manager = (DownloadManager) mContext.getSystemService(Context.DOWNLOAD_SERVICE);
+        manager.remove(mDownloadInfo.get(position).getDownloadId());
```

Comment:
Also remove it from our own DB so that we won't see a "UNKNOWN" item after it is canceled. Or maybe we can add another canceled state which can only be removed but not deleted. I have pinged @FlyTori for further clarification.

Reply:
Let's remove the item when it is deleted.

Reply:
So when user cancel the download, we won't record in our DB?

Reply:
I think whenever we see a STATUS_FAILED let's remove the item directly for simplicity. If you feel like leaving the item there and showing a canceled status would be simpler/better, just let me know and I'll consult @FlyTori . (Or contact here directly :3)

Reply:
Update: After tracing the code, it actually shouldn't be STATUS_FAILED. If a download is canceled, it will not have an entry in the DownloadManager so we should remove it if we failed to query. See: https://github.com/mozilla-tw/Zerda/blob/2efc0e015a8195077fb8925785f2e5a9822bc176/app/src/main/java/org/mozilla/focus/download/DownloadInfoManager.java#L255

Reply:
resolved